std/typecons.d

```diff
@@ -149,25 +157,37 @@ public:
     ~this()
     {
         debug(Unique) writeln("Unique destructor of ", (_p is null)? null: _p);
-        if (_p !is null) delete _p;
-        _p = null;
+        release();
     }
-    /** Returns whether the resource exists. */
-    @property bool isEmpty() const
-    {
-        return _p is null;
-    }
-    /** Transfer ownership to a $(D Unique) rvalue. Nullifies the current contents. */
-    Unique release()
+
+    /** Frees the underlying $(D RefT) and nulls it */
+    void release()
     {
         debug(Unique) writeln("Release");
-        auto u = Unique(_p);
-        assert(_p is null);
-        debug(Unique) writeln("return from Release");
-        return u;
+        if (_p !is null)
+        {
+            import core.memory : GC;
+
+            destroy(_p);
+            GC.free(cast(void*)_p);
```

Comment:
From my experimentation in picking apart `emplace`, the startup sequence would have to be:

1. Blit the allocated memory with `typeid(T).init`

2. Somehow assign the frame/context pointer

3. Call the constructor (Doing so without step 2, as `emplace` currently does, causes segfaults when a constructor tries to reach into the context via a null frame pointer.)

I'm all for that option, but how can it be done? @MartinNowak - you mention some way to set the context pointer - could you show this with a small code sample? How does one get the address of their current frame?